macros/sql/surrogate_key.sql

```diff
@@ -0,0 +1,16 @@
+{% macro surrogate_key(fields) -%}
+
+md5(concat(
+
+    {%- for field in fields %}
+
+        coalesce(
+            {{dbt_utils.safe_cast(field, dbt_utils.type_string())}}
```

Comment:
I was initially opposed to using `safe_cast` here because it's possible that two distinct lists of `fields` values would both evaluate to `md5('')` if none of the arguments are cast-able. It looks to me though like pretty much everything can be casted as a string, and `null` is only returned for more restrictive types like `int64` on bigquery.